### DISCUSSION TITLE
Example: 5 Patterns for Callouts and Codeblocks in Callouts

### DISCUSSION THREAD

**Discussion by aus10code** [jglev/obsidian-apply-patterns-plugin#57] [Example Patterns]
## 5 Patterns for Callouts and Codeblocks in Callouts

**Pattern 1: html tag**
Code blocks inside of call outs aren't treated as code blocks in Source/Edit View. This has the unfortunate flaw of `<html>` tags nested in codeblocks inside of callouts being rendered as html.

This pattern adds a `\` before any html tag that could cause an issue in the selected text

**Pattern 2: html tag remove**
Removes the `\` from any code block that has been formatted with Pattern 1 within the selected text

**Pattern 3: Add >**
Adds the `>` character to the beginning of selected text. Purpose is to create callouts more easily

**Pattern 4: Remove >**
Removes the `>` character from the beginning of selected text. Purpose is dismantle callouts more easily

**Pattern 5: Add > Before Char**
Purpose of this pattern is to aid in creating nested callouts inside of bulleted/numbered lists.

This pattern adds `>` after the last `Tab` character at the beginning of every line in the selected text. I've attached a gif of it working in action and how to use it.
![pattern5](https://user-images.githubusercontent.com/62197117/199084426-ff976dfd-d88f-4923-b7a6-a7b94f1df17d.gif)

To correctly use this pattern, you must tab(indent) the unformatted callout the desired amount of tab(indent) spaces for it to fit into the bulleted/numbered list and then apply the pattern.

## JSON
```json
[
  {
    "name": "html tag",
    "rules": [
      {
        "from": "<(?!.*?(?:\\/))(?!!--)",
        "to": "\\<",
        "caseInsensitive": false,
        "global": true,
        "multiline": true,
        "sticky": false,
        "disabled": false
      }
    ],
    "collapsed": false,
    "cursorRegexStart": "$",
    "cursorRegexEnd": "^"
  },
  {
    "name": "html tag remove",
    "rules": [
      {
        "from": "\\\\<",
        "to": "<",
        "caseInsensitive": false,
        "global": true,
        "multiline": true,
        "sticky": false,
        "disabled": false
      }
    ],
    "collapsed": false,
    "cursorRegexStart": "$",
    "cursorRegexEnd": "^"
  },
  {
    "name": "Add >",
    "rules": [
      {
        "from": "^",
        "to": ">",
        "caseInsensitive": false,
        "global": true,
        "multiline": true,
        "sticky": false,
        "disabled": false
      }
    ],
    "collapsed": false,
    "cursorRegexStart": "$",
    "cursorRegexEnd": "^"
  },
  {
    "name": "Remove >",
    "rules": [
      {
        "from": "^>",
        "to": "",
        "caseInsensitive": false,
        "global": true,
        "multiline": true,
        "sticky": false,
        "disabled": false
      }
    ],
    "collapsed": false,
    "cursorRegexStart": "$",
    "cursorRegexEnd": "^"
  },
  {
    "name": "Add > Before Char",
    "rules": [
      {
        "from": "^((.*?)([^\\t]))",
        "to": "$2>$3",
        "caseInsensitive": false,
        "global": true,
        "multiline": true,
        "sticky": false,
        "disabled": false
      }
    ],
    "collapsed": false,
    "cursorRegexStart": "$",
    "cursorRegexEnd": "^"
  }
]
```

**Comment by jglev** [comment]
Thank you for sharing!
